src/components/PostsList.tsx

```diff
@@ -2,85 +1,73 @@
+import React, { useState } from 'react';
+import { Post } from '../types/Post';
+import cl from 'classnames';
 
-export const PostsList: React.FC = () => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+interface Props {
+  posts: Post[];
+  onActivePost: (v: Post | null) => void;
+  onShowSideBar: (v: boolean) => void;
+}
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <th> </th>
-        </tr>
-      </thead>
+export const PostsList: React.FC<Props> = ({
+  posts,
+  onActivePost,
+  onShowSideBar,
+}) => {
+  const [activePost, setActivePost] = useState<Post | null>(null);
+  const [showClose, setShowClose] = useState(false);
 
-      <tbody>
-        <tr data-cy="Post">
-          <td data-cy="PostId">17</td>
+  const handleChangeButton = (post: Post) => {
+    if (post.id === activePost?.id) {
+      setShowClose(!showClose);
+      onShowSideBar(showClose ? false : true);
 
-          <td data-cy="PostTitle">
-            fugit voluptas sed molestias voluptatem provident
-          </td>
+      return;
+    }
 
-          <td className="has-text-right is-vcentered">
-            <button
-              type="button"
-              data-cy="PostButton"
-              className="button is-link is-light"
-            >
-              Open
-            </button>
-          </td>
-        </tr>
+    onActivePost(post);
+    onShowSideBar(true);
 
-        <tr data-cy="Post">
-          <td data-cy="PostId">18</td>
+    setShowClose(true);
+    setActivePost(post);
```

Comment:
Ensure that the logic for setting the active post and showing the sidebar is consistent and doesn't lead to unexpected behavior when switching between posts.